### PR TITLE
Piper/fix block tests invalid state

### PR DESCRIPTION
**Pull request by pipermerriam**
Fix the VM issues and some other bytes/str/int/long py2/3 related bugs.

leaves-6c86c4bf1de24fb4822f97a7b86a8830

### CODE REVIEW

**Comment by pipermerriam** [comment]
@vbuterin this PR fixes some python3 errors that have been introduced in some of your recent work implementing the various EIP proposals.

File: ethereum/utils.py [hunk 1]
```diff
@@ -428,17 +426,6 @@ def __init__(self):
 trie_root = Binary.fixed_length(32, allow_empty=True)
 
 
-class bcolors:
```
Comment (konradkonrad) [inline ethereum/utils.py]: why did you remove this?

Comment (pipermerriam) [inline ethereum/utils.py]: It is only used by the `slogging` module so I moved it over to there.  Happy to move it back if it's a problem.

File: ethereum/slogging.py [hunk 2]
```diff
@@ -20,6 +20,33 @@
...
+
+class bcolors:
+    HEADER = '\033[95m'
+    OKBLUE = '\033[94m'
+    OKGREEN = '\033[92m'
+    WARNING = '\033[91m'
+    FAIL = '\033[91m'
+    ENDC = '\033[0m'
+    BOLD = '\033[1m'
+    UNDERLINE = '\033[4m'
```
Comment (pipermerriam) [inline ethereum/slogging.py]: @konradkonrad it lives here now

File: ethereum/slogging.py [hunk 3]
```diff
@@ -1,9 +1,9 @@
+import sys
 import logging
 import json
 import textwrap
 from json.encoder import JSONEncoder
 from logging import StreamHandler, Formatter, FileHandler
-from ethereum.utils import bcolors, isnumeric
```
Comment (pipermerriam) [inline ethereum/slogging.py]: @konradkonrad removing these two imports makes this module independent from the rest of the package.

File: ethereum/slogging.py (hunk 2 above)
Comment (konradkonrad) [inline ethereum/slogging.py]: ah alright then, I missed that!

**Comment by konradkonrad** [comment]
This looks okay to me -- in general I think I prefer PR's that stay inside their scope, i.e. fixing py3 things should not try to refactor the package structure, too. (Not that I haven't broken that rule myself in the past ;) )

**Comment by pipermerriam** [comment]
I completely agree with keeping pull requests contained to single things. I
feel like this specific refactor was somehow related but I'm struggling to
recall how. Maybe relative imports?

 Either way I'll do better keeping unrelated things separate
